pkg/ingester/ingester_v2.go

```diff
@@ -799,13 +801,20 @@ func (i *Ingester) v2Push(ctx context.Context, req *cortexpb.WriteRequest) (*cor
 				continue
 			}
 
-			var ve *validationError
-			if errors.As(cause, &ve) {
-				// Caused by limits.
-				if firstPartialErr == nil {
-					firstPartialErr = ve
+			if cause == errMaxSeriesPerUserLimitExceeded || cause == errMaxSeriesPerMetricLimitExceeded {
```

Comment:
We can simplify the logic on lines 787 – 820 with:

```
		updateFirstPartial := func(errFn func() error) {
			if firstPartialErr == nil {
				firstPartialErr = errFn()
			}
		}

		switch cause {
		case storage.ErrOutOfBounds:
			sampleOutOfBoundsCount++
			updateFirstPartial(func() error { return wrappedTSDBIngestErr(err, model.Time(s.TimestampMs), ts.Labels) })
			continue
		case storage.ErrOutOfOrderSample:
			sampleOutOfOrderCount++
			updateFirstPartial(func() error { return wrappedTSDBIngestErr(err, model.Time(s.TimestampMs), ts.Labels) })
			continue
		case storage.ErrDuplicateSampleForTimestamp:
			newValueForTimestampCount++
			updateFirstPartial(func() error { return wrappedTSDBIngestErr(err, model.Time(s.TimestampMs), ts.Labels) })
			continue
		case errMaxSeriesPerUserLimitExceeded:
			perUserSeriesLimitCount++
			updateFirstPartial(func() error { return makeLimitError(perUserSeriesLimit, i.limiter.FormatError(userID, cause)) })
			continue
		case errMaxSeriesPerMetricLimitExceeded:
			perMetricSeriesLimitCount++
			updateFirstPartial(func() error { return makeMetricLimitError(perMetricSeriesLimit, copiedLabels, i.limiter.FormatError(userID, cause))	})
			continue
		}
```

Reply:
Good idea, done.